### PR TITLE
BTRX - 228 - Add HMB & POA terms & logic to Data Access Request Application

### PR DESCRIPTION
**Pull request by LeoForconesi**
**_Description_**
This code maps Poa and Hmb values to be received by ontology to make translate logic.
Also this code has its Ui part which can be seen in [BTRX-228](https://github.com/DataBiosphere/consent-ui/pull/187)

**_Jira Ticket_**
[BTRX-228-Add HMB & POA terms & logic to Data Access Request Application](https://broadinstitute.atlassian.net/browse/BTRX-228)

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: consent-ws/src/main/java/org/broadinstitute/consent/http/service/UseRestrictionConverter.java [hunk 1]
```diff
@@ -88,6 +88,20 @@ else if (selectedGender.equalsIgnoreCase("F"))
         if (pediatricsOnly) {
             dataUseDTO.setPediatric(true);
         }
+
+
+        if(Boolean.valueOf(form.getOrDefault("poa", false).toString())){
```
Comment (rushtong) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/service/UseRestrictionConverter.java]: Nit spacing throughout. Please check your IDE settings to ensure that the space settings are consistent with the rest of the code. For example, this line should be:
```
if (Boolean.valueOf(form.getOrDefault("poa", false).toString())) {
```

**Review by rushtong — APPROVED** [review]
👍 Looks good, modulo minor formatting issues. If you resolve those, I'm good.

File: consent-ws/src/main/java/org/broadinstitute/consent/http/service/UseRestrictionConverter.java (hunk 1 above)
Comment (LeoForconesi) [reply consent-ws/src/main/java/org/broadinstitute/consent/http/service/UseRestrictionConverter.java]: Sorry for that, this is fixed now

File: consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java [hunk 2]
```diff
@@ -170,6 +170,14 @@ public boolean isSensitivePopulation() {
             researchList.add(new SummaryItem(SummaryConstants.RT_POPULATION, SummaryConstants.RT_POPULATION_DETAIL));
             manualReviewIsTrue();
         }
+        if(darDocument.containsKey("hmb") && darDocument.getBoolean("population")){
```
Comment (ahaessly) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java]: should be getBoolean("hmb")?

File: consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java [hunk 3]
```diff
@@ -170,6 +170,14 @@ public boolean isSensitivePopulation() {
...
+            researchList.add(new SummaryItem(SummaryConstants.RT_HEALTH_BIOMEDICAL, SummaryConstants.RT_HEALTH_BIOMEDICAL_DETAIL));
+            manualReviewIsTrue();
+        }
+        if(darDocument.containsKey("poa") && darDocument.getBoolean("population")){
```
Comment (ahaessly) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java]: should be getBoolean("poa")

**Review by ahaessly — CHANGES_REQUESTED** [review]
see changes

File: consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java (hunk 2 above)
Comment (rushtong) [reply consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java]: great catches!

**Review by ahaessly — APPROVED** [review]
👍

File: consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java (hunk 3 above)
Comment (LeoForconesi) [reply consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java]: Whoa i really have missed those. Fixed!

**Comment by ahaessly** [comment]
are there tests to add for this?

**Comment by LeoForconesi** [comment]
At this point, there isn't any.
Later when we use these values for summaries, and reports I'm sure we'll add tests including them, but for this instance we are only mapping values to be used by ontology. Does it sounds ok for you @ahaessly?

**Comment by ahaessly** [comment]
ok. thanks for explaining

**Pull request merged**